### PR TITLE
SONAR-6702 A component key should by unique in database

### PR DESCRIPTION
**Pull request by julienlancelot**

### CODE REVIEW

**Comment by simonbrandhof** [comment]
The db index on PROJECTS[KEE, ENABLED] must be replaced by an UNIQUE index on KEE only.

**Comment by julienlancelot** [comment]
@simon feedback applied !

File: server/sonar-server/src/main/java/org/sonar/server/db/migrations/v52/RemoveDuplicatedComponentKeys.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+    @Override
+    public Component read(Select.Row row) throws SQLException {
+      return new Component(row.getLong(1), row.getString(2), row.getString(3), row.getBoolean(4));
+    }
+  }
+
+  private static class Component {
+    private final long id;
+    private final String uuid, projectUuid;
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/db/migrations/v52/RemoveDuplicatedComponentKeys.java]: MINOR Declare "projectUuid" on a separate line. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1659)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MINOR 1 minor

Watch the comments in this conversation to review them.

**Comment by julienlancelot** [comment]
@simonbrandhof, last feedback has been applied

**Comment by simonbrandhof** [comment]
LGTM

File: server/sonar-server/src/main/java/org/sonar/server/db/migrations/v52/RemoveDuplicatedComponentKeys.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+    @Override
+    public Component read(Select.Row row) throws SQLException {
+      return new Component(row.getLong(1), row.getString(2), row.getString(3), row.getBoolean(4));
+    }
+  }
+
+  private static class Component {
+    private final long id;
+    private final String uuid;
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/db/migrations/v52/RemoveDuplicatedComponentKeys.java]: MINOR Declare "projectUuid" on a separate line. [rule](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1659)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue:
* MINOR 1 minor

Watch the comments in this conversation to review them.

**Pull request merged**